DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js

```diff
@@ -616,32 +618,29 @@ function determineValidity() {
       return false;
     }
   }
-
+  if (!checkDateValidity()) {
+    alert('Start date cannot be after or the same day as end date!');
+  }
   return checkDateValidity();
 }
 
 function checkDateValidity() {
-  var startDate = document.getElementById('startDate').value.split('-');
-  var endDate = document.getElementById('endDate').value.split('-');
+  let startDate = document.getElementById('startDate').value.split('-');
+  let endDate = document.getElementById('endDate').value.split('-');
   const year = 0;
   const month = 1;
   const day = 2;
-  let startDateObject = new Date();
-  let endDateObject = new Date();
-
-  startDateObject.setFullYear(parseInt(startDate[year]));
-  startDateObject.setMonth(parseInt(startDate[month]));
-  startDateObject.setDate(parseInt(startDate[day]));
-
-  endDateObject.setFullYear(parseInt(endDate[year]));
-  endDateObject.setMonth(parseInt(endDate[month]));
-  endDateObject.setDate(parseInt(endDate[day]));
-
-  if (endDateObject <= startDateObject) {
+  startDate[month] = startDate[month] < 10 ? '0' + startDate[month] : startDate[month];
+  startDate[day] = startDate[day] < 10 ? '0' + startDate[day] : startDate[day];
+  
+  startDate = new Date(`${startDate[month]}/${startDate[day]}/${startDate[year]}`);
```

Comment:
Can you change the variable names? You are using the same variable names for a lot of stuff (e.g. startDate for the document element and startDate for the Date object) and this was one of the bugs I had to fix this morning.

Reply:
Fixed